src/browser/pending_subscriptions.ts

```diff
@@ -0,0 +1,169 @@
+        applyPendingSubscription(runtime, subscription);
+    });
+}
+
+/**
+ * Subscribe to an event in a remote runtime
+ */
+function applyPendingSubscription(runtime: PeerRuntime, subscription: IPendingSubscription) {
+    const classEventEmitter = getClassEventEmitter(runtime, subscription);
+    const runtimeVersion = runtime.portInfo.version;
+    const { uuid, name, className, eventName, listenType } = subscription;
+    const fullEventPath = `${className}/${eventName}/${uuid}-${name}`;
+
+    function listener(data: any) {
+        ofEvents.emit(fullEventPath, data);
+
+        // As soon as the event listener fires, we know which runtime is a true
+        // subscription 'holder', all other runtimes should remove this subscription
+        cleanUpSubscription(subscription, runtime.portInfo.version);
+    }
+
+    // Subscribe to an event on a remote runtime
+    classEventEmitter[listenType](eventName, listener);
+
+    // Store a cleanup function for the added listener in
+    // un-subscription map, so that later we can remove extra subscriptions
+    subscription.unSubscriptions[runtimeVersion] = () => {
+        classEventEmitter.removeListener(eventName, listener);
+    };
+}
+
+/**
+ * Clean up a pending subscription.
+ *
+ * Clean up is done when one of the listener fires. This means at that moment
+ * we know which runtime is a true subscriptions holder. We then remove the
+ * subscription from other runtimes and from the pending list.
+ */
+function cleanUpSubscription(subscription: IPendingSubscription, keepInRuntimeVersion?: string) {
+
+    // Already been cleaned before in unneeded runtime versions
+    if (subscription.isCleaned && keepInRuntimeVersion) {
+        return;
+    }
+
+    // Cleanup subscriptions in all connected runtimes
+    connectionManager.connections.forEach((conn) => {
+        const runtimeVersion = conn.portInfo.version;
+        const runtimeUnSubscription = subscription.unSubscriptions[runtimeVersion];
+
+        // Don't un-subscribe in the runtime where we need to keep the subscription
+        if (runtimeVersion === keepInRuntimeVersion) {
+            return;
+        }
+
+        // Invoke un-subscribe function for runtimes that have that subscription
+        if (typeof runtimeUnSubscription === 'function') {
+            runtimeUnSubscription();
+        }
+    });
+
+    // Remove subscription from the list of pending subscriptions
+    const subscriptionIndex = pendingSubscriptions.findIndex(s => s._id === subscription._id);
+    pendingSubscriptions.splice(subscriptionIndex, 1);
+
+    // Indicate the cleaning is done, so that we don't unnecessarily repeat the cleanup
+    subscription.isCleaned = true;
+}
+
+/**
+ * Get event emitter of the class
+ */
+function getClassEventEmitter(runtime: PeerRuntime, subscription: IPendingSubscription) {
+    let classEventEmitter;
+    const { uuid, name, className } = subscription;
+
+    switch (className) {
```

Comment:
Not needed, because it's an internal call and it's TypeScript ;) (above shape allows only certain class names `className: 'application'|'window'`, etc..), so `default` should/will never happen